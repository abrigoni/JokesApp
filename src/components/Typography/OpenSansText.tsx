import React from 'react';
import {Text, StyleSheet} from 'react-native';

const FONT_NAME = 'OpenSans';
type TextVariants = 'Bold' | 'Regular' | 'SemiBold';
type TextSizes = 'H1' | 'H2' | 'Body';

type OpenSansTextProps = {
  children: React.ReactNode;
  variant?: TextVariants;
  size?: TextSizes;
  onPress?: () => void;
  style?: any;
}

const sizesStyle = StyleSheet.create({
  H1: {
    fontSize: 30,
  },
  H2: {
    fontSize: 24,
  },
  Body: {
    fontSize: 17,
  },
});

const variantsStyle = StyleSheet.create({
  Bold: {
    fontFamily: `${FONT_NAME}-Bold`,
  },
  SemiBold: {
    fontFamily: `${FONT_NAME}-SemiBold`,
  },
  Regular: {
    fontFamily: `${FONT_NAME}-Regular`,
  },
});

const OpenSansText = ({
  children,
  variant = 'Regular',
  size = "Body",
  onPress,
  style,
}: OpenSansTextProps) => {
  return (
    <Text style={[sizesStyle[size], variantsStyle[variant], style]} onPress={onPress}>
      {children}
    </Text>
  );
};

export default OpenSansText;
