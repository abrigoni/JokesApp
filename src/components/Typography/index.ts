export { default as OpenSansText } from './OpenSansText';
