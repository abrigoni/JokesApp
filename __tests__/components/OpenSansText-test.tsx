import 'react-native';
import React from 'react';
import {OpenSansText} from '../../src/components/Typography';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<OpenSansText />', () => {
  it('renders correctly', () => {
    const json = renderer
      .create(
        <OpenSansText size={'Body'} variant={'Bold'}>
          Test
        </OpenSansText>,
      )
      .toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders correctly without variant', () => {
    const json = renderer
      .create(<OpenSansText size={'Body'}>Test</OpenSansText>)
      .toJSON();
    expect(json).toMatchSnapshot();
  });
});
