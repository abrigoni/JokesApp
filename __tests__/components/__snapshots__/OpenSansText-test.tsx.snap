// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpenSansText /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 17,
      },
      Object {
        "fontFamily": "OpenSans-Bold",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`<OpenSansText /> renders correctly without variant 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 17,
      },
      Object {
        "fontFamily": "OpenSans-Regular",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
